### PR TITLE
fix(transformers): Ability to transform `any` with TypeTransformer

### PR DESCRIPTION
**Pull request by disq**
plus testing fixes.

Without this there was no way to transform `any` (unless it's defined as `[]any`) because bare interfaces were ignored in `isTypeIgnored()` along with channels/funcs/unsafeptrs.

### CODE REVIEW

File: transformers/struct.go [hunk 1]
```diff
@@ -239,6 +239,9 @@ func (t *structTransformer) addColumnFromField(field reflect.StructField, parent
 			return fmt.Errorf("failed to transform type for field %s: %w", field.Name, err)
 		}
 	}
+	if columnType == nil {
+		return nil // ignored
```
Comment (disq) [inline transformers/struct.go:243]: This ensures backwards compatibility with the current situation where we ignore `reflect.Interface` early in `isTypeIgnored`.

File: transformers/struct.go [hunk 2]
```diff
@@ -348,6 +350,9 @@ func defaultGoTypeToSchemaType(v reflect.Type) (arrow.DataType, error) {
 		}
 		return arrow.ListOf(elemValueType), nil
 
+	case reflect.Interface:
+		return nil, nil // silently ignore
```
Comment (disq) [inline transformers/struct.go:354]: This ensures we don't panic now that we don't ignore this kind in `isTypeIgnored`.

File: transformers/struct_test.go [hunk 3]
```diff
@@ -405,11 +459,21 @@ func TestTableFromGoStruct(t *testing.T) {
...
+
 			for i, col := range table.Columns {
 				if !arrow.TypeEqual(col.Type, tt.want.Columns[i].Type) {
-					t.Fatalf("column %s does not match expected type. got %v, want %v", col.Name, col.Type, tt.want.Columns[i].Type)
+					t.Fatalf("column %q does not match expected type. got %v, want %v", col.Name, col.Type, tt.want.Columns[i].Type)
+				}
+			}
+			for _, exc := range tt.want.Columns {
+				if c := table.Column(exc.Name); c == nil {
+					t.Fatalf("column %q not found. want: %v", exc.Name, exc.Type)
```
Comment (hermanschaaf) [inline transformers/struct_test.go:473]: Nice 👍

**Review by hermanschaaf — APPROVED** [review]

**Comment by disq** [comment]
Ran this on _all_ source plugins - no changes.

**Review by yevgenypats — APPROVED** [review]

**Pull request merged**